Improvements in concurrency and tailwindcss integration

acemavrick opened the pull request:
Main changes:
* Made scraping concurrent
* Added an option to use local tailwindcss installations

I have attempted to make sure everything is in sync with all commits on the main repository up to [this commit](https://github.com/warithr621/uil-hub/commit/fdaa8cc0b9441e2e566f21a38801baeeb6f8ecd1).

acemavrick committed:
integrated tailwindcss via npm

Switched from CDN to local

acemavrick committed:
Made scraping results concurrent

acemavrick committed:
Merge upstream changes while maintaining concurrent processing

warithr621 commented:
Can you recheck the team outputs? It seems like the prog score is getting printed out for non-prog events.

<img width="1498" alt="image" src="https://github.com/user-attachments/assets/048020b9-7815-41a7-8c7f-4a99d4597961" />

<img width="1498" alt="image" src="https://github.com/user-attachments/assets/095a1500-4452-4788-9b85-13cf9cc258e5" />

<img width="1509" alt="image" src="https://github.com/user-attachments/assets/2ee9f3fe-edb4-428e-b107-8d23641c3c72" />

<img width="1513" alt="image" src="https://github.com/user-attachments/assets/f6fb5edf-6725-4581-a6aa-ef9d2f765a23" />

As you can see, all the non-CS events have the -999 printing out in the team results. Additionally, some of the column headers are completely gone, such as the Bio/Chem/Phys one and the Prog one that would usually be there for CS.
Also make sure to check the merge conflict issue if you can, it mentions a merge conflict with `app.py`... :(

Admittedly I could probably fix this on my own but ehhh you got it.

warithr621 reviewed (changes requested):
looks rly good, just refer to comment in "Conversation" tab (i.e. right above this) for changes pls 👍

acemavrick committed:
Bug Fixes

- Fixed headers not displaying for Science & CS Programming scores
- Fixed incorrect display of `-999` for non-CS team scores

acemavrick committed:
Merge branch 'main' into main
